[cppia] Generate full debug source paths

tobil4sk opened the pull request:
See #12052

Perhaps this should be on by default, given that the debugger doesn't work without this, but I'm not sure if doing that might break anything else.

tobil4sk committed:
[cppia] Allow generating full debug source paths

Aidan63 commented:
I think this should be controlled by the existing `absolute-path` define. As of #11763 setting that will cause stack frames to point to the absolute files and defining it places absolute paths in cppia files.

![image](https://github.com/user-attachments/assets/5ad03cfd-ad26-4c8d-913e-0c2771fdd8e5)

Aidan63 commented:
As for enabling by default, I think it makes sense that these sort of paths are absolute if the debug flag is set.

tobil4sk commented:
> I think this should be controlled by the existing absolute-path define. 

I saw absolute-path but dismissed it after reading the documentation, since it refers to trace output: https://github.com/HaxeFoundation/haxe/blob/7ba8e17cfb78be8d6f812af6df9fdc8fc2140ca6/src-json/define.json#L5

Unlike `-D absolute-path`, this new behaviour leaves trace untouched. My intention is just to fix the debugger, which shouldn't affect trace output.

Aidan63 commented:
That doc comment should probably be updated then, it already does far more than that on the cpp target.

tobil4sk commented:
Does it definitely make sense to make use of `absolute-path` in #11763? It doesn't seem like it is ever used for a purpose like this on any other target.

Also, your change means that `-D absolute-path` results in `cpp.vm.Debugger.getFiles()` and `cpp.vm.Debugger.getFilesFullPath()` both containing the same paths. Since the full paths are always available in hxcpp anyway, perhaps there is a way to get the stack frames to use those full paths without affecting `getFiles()`?

Simn commented:
If the debugger doesn't work without this then I don't see why we shouldn't make it the default. In fact, in that case I don't see why we'd even need a define to change it to the previous behavior.

tobil4sk committed:
Use full debug paths in cppia in all cases

Without these paths being absolute, cppia break points cannot work at
all, so it makes more sense for this to just be the default. These paths
are only ever added for debugging anyway.

tobil4sk commented:
> If the debugger doesn't work without this then I don't see why we shouldn't make it the default. In fact, in that case I don't see why we'd even need a define to change it to the previous behavior.

I've made it so it always uses the full paths. I guess if there is ever an issue with the behaviour then we can add a flag so it's possible to go back to the old behaviour.
